### PR TITLE
Fix button bitmask being used as a button count, making gamepad mappings impossible

### PR DESCRIPTION
**Pull request by CookiePLMonster**
Port from:
https://github.com/CookiePLMonster/SilentPatchNFS90s/blob/a25a586f83e5ba3e4c82eb23c6fe2e32e07fc313/source/SilentPatchNFS90s.cpp#L619-L627

See https://github.com/zaps166/NFSIISE/issues/104 for a discussion and explanation of this fix.

**WARNING: Untested!**

### CODE REVIEW

File: NFS2SE.asm [hunk 1]
```diff
@@ -42015,8 +42015,7 @@ loc_421DE3:
 	mov cl, bl
```
Comment (zaps166) [inline NFS2SE.asm:42015]: I think these 2 lines could be removed now:
```asm
mov cl, bl
mov edi, 1
```

I added offset in C code to simulate joystick with more buttons and It's working!

Comment (zaps166) [reply NFS2SE.asm:42015]: I removed these 2, too. Let's hope it'll work without them :sweat_smile:

**Comment by zaps166** [comment]
Thanks!

File: NFS2SE.asm (hunk 1 above)
Comment (CookiePLMonster) [reply NFS2SE.asm:42015]: True, good catch! I didn't bother deleting them in my standalone patch so I patch as little as possible, but here it makes perfect sense to get rid of them.

**Pull request merged**